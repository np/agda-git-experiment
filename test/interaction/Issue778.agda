-- {-# OPTIONS -v tc.term.exlam:100 -v extendedlambda:100 -v int2abs.reifyterm:100 #-}
-- Andreas, 2013-02-26
module Issue778 (Param : Set) where

data ⊥ : Set where

interactive : ⊥ → ⊥
interactive = λ {x → {!x!}}
  where aux : ⊥ → ⊥
        aux x = x

-- splitting did not work for extended lambda in the presence of
-- module parameters and where block

data ⊤ : Set where
  tt : ⊤

test : ⊤ → ⊤
test = λ { x → {!x!} }
  where aux : ⊤ → ⊤
        aux x = x
