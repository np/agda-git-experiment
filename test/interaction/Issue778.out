Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue778 (Issue778.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue778.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : ⊥\n?1 : ⊤\n" nil)
((last . 1) . (agda2-goals-action '(0 1)))
Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("()")))
((last . 1) . (agda2-goals-action '(0 1)))
Agda2> ((last . 2) . (agda2-make-case-action-extendlam '("tt → ?")))
((last . 1) . (agda2-goals-action '(0 1)))
Agda2> (agda2-give-action 1 "tt")
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : ⊥\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> 